front/tests/utils/scenario.ts

```diff
@@ -14,29 +14,61 @@ interface SetupResult {
   timetableResult: TimetableResult;
 }
 
-// Define the setupScenario function
-export default async function setupScenario(
+/**
+ * Sets up a scenario by fetching required infrastructure, project, study, and creating a new scenario.
+ *
+ * @param {number | null} [electricalProfileId=null] - Optional electrical profile ID for the scenario.
+ * @returns {Promise<SetupResult>} - The setup result containing the infrastructure, project, study, scenario, and timetable result.
+ */
+export default async function createScenario(
+  projectId: number | null = null,
+  studyId: number | null = null,
+  infraId: number | null = null,
   electricalProfileId: number | null = null
 ): Promise<SetupResult> {
-  // Fetch infrastructure, project, study, and timetable result
-  const smallInfra = await getInfra('small_infra_test_e2e');
-  const project = await getProject('project_test_e2e');
-  const study = await getStudy(project.id, 'study_test_e2e');
+  // Fetch or create infrastructure
+  let smallInfra: Infra;
+  if (!infraId) {
+    smallInfra = await getInfra();
+  } else {
+    smallInfra = { id: infraId } as Infra;
+  }
+
+  // Fetch or create project
+  let project: Project;
+  if (!projectId) {
+    project = await getProject();
+  } else {
+    project = { id: projectId } as Project;
+  }
+
+  // Fetch or create study
+  let study: Study;
+  if (!studyId) {
+    study = await getStudy(project.id);
+  } else {
+    study = { id: studyId } as Study;
+  }
+
+  // Create a new timetable result
   const timetableResult = await postApiRequest(`/api/timetable/`);
 
-  // Create a new scenario with a unique name
-  const scenario = await postApiRequest(
+  // Create a new scenario with a unique name using UUID
+  const scenario: Scenario = await postApiRequest(
     `/api/projects/${project.id}/studies/${study.id}/scenarios/`,
     {
       ...scenarioData,
-      name: `${scenarioData.name} ${uuidv4()}`,
+      name: `${scenarioData.name} ${uuidv4()}`, // Generating a unique name
       study_id: study.id,
       infra_id: smallInfra.id,
       timetable_id: timetableResult.timetable_id,
       electrical_profile_set_id: electricalProfileId,
-    }
+    },
+    undefined,
+    'Failed to create scenario'
```

Comment:
Is it really useful to have these messages? Isn't the request method + path enough to figure out what the request is doing?

Reply:
It's not necessary, it's just that the specific message can add clarity without adding significant overhead